python_modules/dagster/dagster/_core/asset_graph_view/asset_graph_view.py

```diff
@@ -295,3 +331,17 @@ def compute_intersection_with_partition_keys(
                 asset_slice.asset_key, partitions_def, partition_keys
             ),
         )
+
+    def compute_asset_slice_with_newly_updated_parents(
+        self, asset_key: AssetKey, cursor: Optional[int]
+    ) -> "AssetSlice":
+        """Computes the slice of the asset that has parents with new materialization events since
+        the cursor.
+        """
+        asset_partitions, _ = (
```

Comment:
note: the `_` here is what originally would be used as the new "safe" maximum record id. In the new world, my understanding is that the plan is to calculate a safe maximum record id before we even invoke this function (so this will be part of the temporal context), and so the underlying method we're invoking below will need to be reworked to take that temporal context into account.